public namespace back3;
import common -;

struct foo {
  string s;
  int x;
  int y;
}

{
  exception::set_stack_trace_limit(0);
  try {
    varray{foo} v;
    v.emplace_back("abc", 10, 20);
    foo const& f = v[0];
    v.emplace_back("xyz", 15, 3); // throw
    println(v);
  } catch (c_exception ex) {
    println("caught", ex);
  }
}

